app/src/test/java/org/pesmypetcare/mypetcare/controllers/exercise/TestTrUpdateExercise.java

```diff
@@ -0,0 +1,138 @@
+package org.pesmypetcare.mypetcare.controllers.exercise;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.pesmypetcare.httptools.utilities.DateTime;
+import org.pesmypetcare.mypetcare.features.pets.Pet;
+import org.pesmypetcare.mypetcare.features.pets.PetRepeatException;
+import org.pesmypetcare.mypetcare.features.pets.events.InvalidPeriodException;
+import org.pesmypetcare.mypetcare.features.pets.events.exercise.Exercise;
+import org.pesmypetcare.mypetcare.features.pets.events.exercise.NotExistingExerciseException;
+import org.pesmypetcare.mypetcare.features.users.NotPetOwnerException;
+import org.pesmypetcare.mypetcare.features.users.User;
+import org.pesmypetcare.mypetcare.services.StubPetManagerService;
+import org.pesmypetcare.usermanager.datacontainers.pet.GenderType;
+
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author Albert Pinto
+ */
+public class TestTrUpdateExercise {
+    private User user;
+    private Pet pet;
+    private TrUpdateExercise trUpdateExercise;
+
+    @Before
+    public void setUp() throws PetRepeatException {
+        user = new User("johnDoe", "johndoe@gmail.com", "1234");
+        pet = getDinkyPet();
+        trUpdateExercise = new TrUpdateExercise(new StubPetManagerService());
+
+        pet.addExercise(new Exercise("Frisbee", "Playing at the beach",
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String "Frisbee" appears 3 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5561369)